### DISCUSSION TITLE
@tophf - Where are the userscripts stored?

### DISCUSSION THREAD

**Discussion by gaveitatry2** [violentmonkey/violentmonkey#1151] [Q&A]
I am using Firefox Portable, and my profile folder is located here:
"G:\009__Other\Firefox Portable\Data\profile"

Inside that profile folder is a ton of folders and files.  It would take too long to list them all or to post screenshots.  Anyway, I'm thinking that since this is a portable version of Firefox Portable that all my userscripts are stored somewhere on my USB drive (where Firefox Portable is installed) and not somewhere on my computer.  I just want to know where they are.

Also, is it possible for me to keep all my userscripts in a specific folder, i.e. "G:\Userscripts" and have them all work in Violentmonkey... but if I ever uninstall Violentmonkey (without exporting settings and userscripts as .zip) my userscripts don't disappear and still stay in "G:\Userscripts"?

**Comment by tophf** [comment]
Extensions can't change the location of storage, see https://stackoverflow.com/questions/17774857/where-is-chrome-extensions-chrome-storage-local-data-saved

**Comment by gaveitatry2** [comment]
Okay, sorry for creating a duplicate.  I couldn't find my post in Closed or Open, so I thought a glitch happened and it didn't get posted.

**Comment by gaveitatry2** [comment]
I don't understand.  I still can't find it.  According to the thread that you posted, the data for Firefox Portable is stored here:
"C:\Users\Brian\AppData\Roaming\Mozilla\Firefox\Profiles\x5jztslz.default\browser-extension-data\"

I got as far as "C:\Users\Brian\AppData\Roaming\Mozilla\Firefox\Profiles\" and there were two profiles.
odopvjvy.default only had one file in it called times.json.
pr3y8mq2.default-release had a bunch of folders and files, but there was no "browser-extension-data" folder.  I don't even know why my extension data would be in there since I am using a portable version of Mozilla Firefox and all data is supposed to be stored on my USB drive, not my AppData folder.

I've asked the developer of Firefox Portable and I've asked you, the developer of Violentmonkey.  And between the two of you, I still don't know where my userscripts are stored.

**Reply by tophf** [reply]
Firefox uses two AppData types: `Roaming` and `Local`. Try the latter.

**Comment by gaveitatry2** [comment]
Actually I tried deleting both Roaming and Local "Mozilla" folders.  Because it is a Portable browser (installed on USB drive) and I knew those folders didn't store the data.  Sure enough they don't.  My userscripts are still there.  But the stupid Firefox Portable developer is wasting my time.  He of all people should know where the data is stored, since it is HIS portable browser.  But he doesn't know.  I give up.  If anyone is to blame, it is him.  He is useless.  Thanks anyway.  https://portableapps.com/node/63254

**Comment by gaveitatry2** [comment]
I think that the data might be stored in the Profile > Storage > Default folder.

Can you tell me the Moz Extension number of ViolentMonkey 2.12.4 (older version)?

**Reply by tophf** [reply]
It didn't change as far as I know.

**Comment by brsk93** [comment]
Hello, I ask here to avoid opening a new issue.
I too need to know where this extension saves the userscripts. I am interested in the actual javascript files, not necessarily stored values.
Today I opened firefox and found out all my scripts were gone and the extension was broken. I have reinstalled it for now but I have lost one month of work since my last full profile backup.
I would like to at least attempt a manual recovery from whatever file was corrupted.
I am using regular Firefox, not the Portable one.

**Reply by tophf** [reply]
The extension uses `browser.storage.local` API for everything including the code. I don't know where Firefox stores it. Try finding articles that show how to recover extension storage from files directly.

**Reply by gera2ld** [reply]
https://violentmonkey.github.io/faq/#where-are-the-userscripts-stored

**Comment by Freed-Wu** [comment]
You can store these scripts in local path. <https://violentmonkey.github.io/posts/how-to-edit-scripts-with-your-favorite-editor/>
